source/FreeRTOS_ARP.c

```diff
@@ -517,107 +510,37 @@ BaseType_t xIsIPInARPCache( uint32_t ulAddressToLookup )
-                                                ( eType == eIPv6_Global ) ? "Global" :
-                                                ( eType == eIPv6_LinkLocal ) ? "LinkLocal" :
-                                                ( eType == eIPv6_Loopback ) ? "Loopback" :
-                                                "other" ) );
-
-                       if( eType == eIPv6_LinkLocal )
-                       {
-                           MACAddress_t xMACAddress;
-                           NetworkEndPoint_t * pxEndPoint;
-                           eARPLookupResult_t eResult;
-                           char pcName[ 80 ];
-
-                           ( void ) memset( &( pcName ), 0, sizeof( pcName ) );
-                           eResult = eNDGetCacheEntry( pxIPAddress, &xMACAddress, &pxEndPoint );
-                           FreeRTOS_printf( ( "xCheckRequiresARPResolution: eResult %s with EP %s\n", ( eResult == eARPCacheMiss ) ? "Miss" : ( eResult == eARPCacheHit ) ? "Hit" : "Error", pcEndpointName( pxEndPoint, pcName, sizeof pcName ) ) );
-
-                           if( eResult == eARPCacheMiss )
-                           {
-                               NetworkBufferDescriptor_t * pxTempBuffer;
-                               size_t uxNeededSize;
-
-                               uxNeededSize = sizeof( ICMPPacket_IPv6_t );
-                               pxTempBuffer = pxGetNetworkBufferWithDescriptor( BUFFER_FROM_WHERE_CALL( 199 ) uxNeededSize, 0U );
-
-                               if( pxTempBuffer != NULL )
-                               {
-                                   pxTempBuffer->pxEndPoint = pxNetworkBuffer->pxEndPoint;
-                                   pxTempBuffer->pxInterface = pxNetworkBuffer->pxInterface;
-                                   vNDSendNeighbourSolicitation( pxTempBuffer, pxIPAddress );
-                               }
-
-                               xNeedsARPResolution = pdTRUE;
-                           }
-                       }
-                   }
-
-                   break;
-               }
-        #endif /* ( ipconfigUSE_IPv6 != 0 ) */
-
-        default:
-            /* Shouldn't reach here */
-            /* MISRA 16.4 Compliance */
-            break;
-    }
+        BaseType_t xNeedsARPResolution = pdFALSE;
+
+        /* MISRA Ref 11.3.1 [Misaligned access] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        const IPPacket_t * pxIPPacket = ( ( const IPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+        const IPHeader_t * pxIPHeader = &( pxIPPacket->xIPHeader );
+        const IPV4Parameters_t * pxIPv4Settings = &( pxNetworkBuffer->pxEndPoint->ipv4_settings );
+
+        /* configASSERT( ( pxIPPacket->xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE ) || ( pxIPPacket->xEthernetHeader.usFrameType == ipARP_FRAME_TYPE ) ); */
```

Comment:
If this is a temporary test code, please remove this line. Or the assert can be uncommented so that it could help with debugging.
I prefer to keep the assert.